### ISSUE TITLE
[HOLD for payment 2024-10-29] [$250] Scan –"Allow location Access" modal is not displayed when submitting a scan expense via QAB

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#50042]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.41-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** applausetester+shsbc11@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition
Make sure that location permission is disabled in your browser

1. Log in to the app with any account
2. Click on FAB > Submit Expense > Scan > Upload a receipt > Choose a recipient > Submit
3. Here notice that "Allow location access" modal is displayed
4. Click on "Not now" option > The expense is created
5. Now click on FAB > then click on QAB > Upload a receipt > Submit
6. Here notice that "Allow location access" modal is not displayed and the expense is created bypassing this process

## Expected Result:
"Allow location access" modal is displayed

## Actual Result:
"Allow location access" modal is not displayed, there is an inconsistency when submitting the expense through FAB and QAB
## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/13186744-af9c-416b-8f02-52a4c6f5086a

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021843470696140504711</li>
        <li>Upwork Job ID: 1843470696140504711</li>
        <li>Last Price Increase: 2024-10-08</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 104325716</li>
<li>nkdengineer | Contributor | 104325723</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @VictoriaExpensify</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by IuliiaHerets** [comment]
@VictoriaExpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"Allow location access" modal is not displayed, there is an inconsistency when submitting the expense through FAB and QAB

### What is the root cause of that problem?

Currently, we skip the confirmation page when a scan expense is created from QBA so it skips "Allow location Access" modal

https://github.com/Expensify/App/blob/ee11bc5ab09c42f6dbe9e0f55e25caddc6d2824a/src/pages/iou/request/step/IOURequestStepScan/index.tsx#L94-L96

### What changes do you think we should make in order to solve the problem?



1. We should add `LocationPermissionModal` to the `IOURequestStepConfirmation` component  


```
          const [isShowLocationPermissionModal, setIsShowLocationPermissionModal] = useState(false);
          const [startLocationPermissionFlow, setStartLocationPermissionFlow] = useState(false);

           {IsShowLocationPermissionModal && (
                <LocationPermissionModal
                    startPermissionFlow={startLocationPermissionFlow}
                    resetPermissionFlow={() => setStartLocationPermissionFlow(false)}
                    onGrant={() => navigateToConfirmationStep(file, source)}
                    onDeny={() => navigateToConfirmationStep(file, source)}
                />
            )}
```

2. Change to logic `navigateToConfirmationStep` function [here](https://github.com/Expensify/App/blob/487816184680744a2ad53a03460cbbdec363c432/src/pages/iou/request/step/IOURequestStepScan/index.tsx#L261) 

```
        const gpsRequired = transaction?.amount === 0 && iouType !== CONST.IOU.TYPE.SPLIT && file;
                 if(shouldSkipConfirmation){
                    const gpsRequired = transaction?.amount === 0 && iouType !== CONST.IOU.TYPE.SPLIT && file;
                    setIsShowLocationPermissionModal(gpsRequired)
                } else {
                    navigateToConfirmationStep(file, source);
                }
```

https://github.com/Expensify/App/blob/487816184680744a2ad53a03460cbbdec363c432/src/pages/iou/request/step/IOURequestStepScan/index.tsx#L261

### What alternative solutions did you explore? (Optional)

**Comment by nkdengineer** [comment]
[Proposal Updated](https://github.com/Expensify/App/issues/50042#issuecomment-2388040634) add minor change and result

**Comment by VictoriaExpensify** [comment]
Reproducible issue. I agree we should fix it

**Comment by DylanDylann** [comment]
The solution is straightforward. @nkdengineer's proposal looks good to me, the detailed implementation will be checked carefully in the PR phase

🎀 👀 🎀 C+ reviewed

**Comment by VictoriaExpensify** [comment]
There is another issue raised that is related to this - [Allow location is not shown if device time is set to 7 days in the future ](https://github.com/Expensify/App/issues/50594)

@DylanDylann @nkdengineer - will your solution resolve this issue?

**Comment by DylanDylann** [comment]
@VictoriaExpensify No, it is another issue

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **It is new implementation, not a bug**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal
1. Click on FAB > Submit Expense > Scan > Upload a receipt > Choose a recipient > Submit
2. Here notice that "Allow location access" modal is displayed
3. Click on "Get It" option > The expense is created
4. Now click on FAB > then click on QAB > Upload a receipt > Submit
5. Verify that: "Allow location access" modal is displayed


Do we agree 👍 or 👎

**Comment by VictoriaExpensify** [comment]
Request raised to add ^^ to testrail - https://github.com/Expensify/Expensify/issues/440022

**Comment by VictoriaExpensify** [comment]
Payment summary updated:
Contributor: @nkdengineer paid $250 via Upwork
Contributor+: @DylanDylann paid $250 via Upwork
